src/ports/realm.ts

```diff
@@ -0,0 +1,145 @@
+import { IBaseComponent } from '@well-known-components/interfaces'
+import { BaseComponents } from '../types'
+import * as fs from 'fs/promises'
+
+export const defaultNames = [
+  'zeus',
+  'poseidon',
+  'athena',
+  'hera',
+  'hephaestus',
+  'aphrodite',
+  'hades',
+  'hermes',
+  'artemis',
+  'thor',
+  'loki',
+  'odin',
+  'freyja',
+  'fenrir',
+  'heimdallr',
+  'baldr'
+]
+
+export type IRealmComponent = IBaseComponent & {
+  getName(lighthouseName?: string): Promise<string | undefined>
+}
+
+export const CATALYST_NAME_CONFIG_FILE = '.catalyst-name'
+
+/*
+  The process of picking a name works as follows:
+
+  1. If there is a file .catalyst-name, just the name there
+  2. If the local lighthouse has a name, pick it, store it in .catalyst-name for next time
+  3. If there is a no name, pick one from the env var REALM_NAMES (or defaultNames if the var is not provided), and ask every catalyst in the DAO if the name is already taken, if it's available, store it in .catalyst-name, otherwise repeat 
```

Comment:
```suggestion
  3. If there is no name, pick one from the list in the env var `REALM_NAMES` (or use `defaultNames` if the var is not provided),  and ask every catalyst in the DAO if the name is already taken, if it's available, store it in .catalyst-name, otherwise repeat 
```

Reply:
If there's no name available, then it will fail to start right? Should we write that too?